### ISSUE TITLE
fmt command gives "semantic difference after formatting" error for multi-line values

### ISSUE THREAD

**Issue by JWFGit** [aws-cloudformation/rain#126]
rain fmt command does not work with multi-line yaml

Using this CF template
```
AWSTemplateFormatVersion: "2010-09-09"
Description: |
  Template generated by rain
  and edited my me
Resources:
  MyFunction:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        ZipFile: |
          import boto3
          import cfnresponse

          def handler(event, context):
              print(event)

      Description: Test lambda for rain fmt
      FunctionName: my-lambda
      Handler: index.handler
      ReservedConcurrentExecutions: 0 # Optional
      Role: arn:aws:iam::123456789012:role/aws-service-role/support.amazonaws.com/AWSServiceRoleForSupport
      Runtime: python3.8

      Tags:
        - Key: my_tag
          Value: ME
      Timeout: 100

```
running ./rain --version

> Rain v1.3.3 darwin/amd64

when the fmt command is run
> ./rain fmt -v my_test/lambda.yaml
```
semantic difference after formatting:
(>) Description: |
  Template generated by rain

  and edited my me
(|) Resources:
(|)   MyFunction:
(|)     Properties:
(|)       Code:
(>)         ZipFile: |
  i        mport boto3
  i        mport cfnresponse
  d        ef handler(event, context):
              print(event)

```

**Comment by ericzbeard** [comment]
I think this is fixed by #117, which hasn't been released yet.

**Comment by ugwis** [comment]
@ericzbeard
I have faced same issue in `rain merge` command at latest commit.

```
$ git log --oneline | head -n 1
e889d62 Merge pull request #130 from ericzbeard/forecast-bucket

$ git show 2af4ccc
commit 2af4cccaf6ea69b11efe2af73c498d1369e77e72
Merge: 4525cb1 163d5e6
Author: Eric Z. Beard <ericzbeard@gmail.com>
Date:   Fri Apr 7 16:10:22 2023 -0700

    Merge pull request #117 from ericzbeard/fix-110
    
    Fix formatting issue with multiline strings

$ cat test.template
Resources:
  CGW:
    Type: AWS::EC2::Instance
    Properties:
      UserData:
        Fn::Base64: !Sub
          - |
            #!/bin/bash -xe
            apt-get update

$ go build ./cmd/rain
$ ./rain merge test.template /dev/null
Resources:
  CGW:
    Type: AWS::EC2::Instance
    Properties:
      UserData: !Base64
        Fn::Sub:
          - |

            #!/bin/bash -xe
            apt-get update
```

I found these multi-line issues caused by go-yaml.v3 generating empty line without indent before formating.
https://github.com/aws-cloudformation/rain/blob/e889d62e5995266e19729538e71a8d994ef0d9c6/internal/cmd/fmt/fmt.go#L43

I tested in go playground.
https://go.dev/play/p/OJt4xbqKWiE
![Screen Shot 2023-05-02 at 0 38 26](https://user-images.githubusercontent.com/914815/235479694-8eba12f1-2995-49c9-a829-6feb2060550c.png)


Not sure where it was specificated in YAML 1.1 spec, but it seems good to follow go.yaml.v3 implementation to treat empty line without indentation as multi-line string.

I create PR.
https://github.com/aws-cloudformation/rain/pull/131

Thanks.

**Comment by ericzbeard** [comment]
@ugwis Thanks for the contribution! This will be included in v1.4
